packages/ui/src/components/molecules/UpdateBar.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+import { Typography, Flex, Image } from '../atoms';
+
+interface UpdateBarProps {
+  text: string;
+  button?: string;
+  handler?: React.MouseEventHandler<HTMLButtonElement>;
+  progress?: number;
+  icon?: string;
+  state?: 'normal' | 'progress' | 'success' | 'error';
+}
+
+const UpdateBarWrap = styled.div`
+  width: 100%;
+  height: ${({ theme }) => theme.spacing.five.spacing};
+  padding-top: ${({ theme }) => theme.spacing.one.spacing};
+  padding-bottom: ${({ theme }) => theme.spacing.one.spacing};
+  padding-left: ${({ theme }) => theme.spacing.two.spacing};
+  padding-right: ${({ theme }) => theme.spacing.two.spacing};
+  border-radius: ${({ theme }) => theme.spacing.one.spacing};
+  background: ${({ theme }) => theme.palette.background.bar};
+  border-color: ${({ theme }) => theme.palette.border.bar};
+  border-width: 1px;
+  border-style: solid;
+`;
+
+const UpdateBarWrapProgress = styled.div<{ progress: number }>`
+  position: absolute;
+  background: ${({ theme }) => theme.palette.golden};
+  border-radius: inherit;
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: ${props => (props.progress > 100 ? 100 : props.progress)}%;
+  transition: 0.1s;
+  z-index: -1;
+`;
+
+const UpdateBarButton = styled.button`
+  background: none;
+  outline: none;
+  border: none;
+  &:hover {
+    filter: brightness(150%);
+    cursor: pointer;
+  }
+`;
```

Comment:
Create a new variant in the existing Button component.

Reply:
sure but ill have to override certain properties like padding and transition